src/main/java/org/broadinstitute/consent/http/service/ReviewResultsService.java

```diff
@@ -48,7 +48,10 @@ public ElectionReview describeElectionReviewByElectionId(Integer electionId, Boo
             review = new ElectionReview();
             review.setElection(election);
             Consent consent = consentDAO.findConsentById(review.getElection().getReferenceId());
-            List<ElectionReviewVote> rVotes = (isFinalAccess == null || isFinalAccess == false) ? voteDAO.findElectionReviewVotesByElectionId(electionId, VoteType.DAC.getValue()) :  voteDAO.findElectionReviewVotesByElectionId(electionId, VoteType.FINAL.getValue());
+            List<ElectionReviewVote> rVotes = isFinalAccess == null ? 
+              voteDAO.findElectionReviewVotesByElectionId(electionId)
+              : isFinalAccess == false ? voteDAO.findElectionReviewVotesByElectionId(electionId, VoteType.DAC.getValue())
+                : voteDAO.findElectionReviewVotesByElectionId(electionId, VoteType.FINAL.getValue());
```

Comment:
The nested ternaries here add a significant degree of complexity that's hard to follow. I'd like to see this broken down so the logic can be read more quickly and cleanly. 

What I think makes this more complex is that we now have 3 operational states of `isFinalAccess`, null, true, or false, and each has different semantics. Using null this way is definitely an anti-pattern. A better approach would be to create an overloaded method with the arguments you want, and then calling that one from the resource class, it might look like this ... 

```
    public ElectionReview describeElectionReviewByElectionId(Integer electionId) {
        ElectionReview review = null;
        Election election = electionDAO.findElectionWithFinalVoteById(electionId);
        if(election != null){
            review = new ElectionReview();
            review.setElection(election);
            Consent consent = consentDAO.findConsentById(review.getElection().getReferenceId());
            List<ElectionReviewVote> rVotes = voteDAO.findElectionReviewVotesByElectionId(electionId);
            review.setReviewVote(rVotes);
            review.setConsent(consent);
        }
        return review;
    }
```

Reply:
yeah good idea I agree and if we want to make this same change for the rp election, so that we are receiving both member and chair votes, then we could get rid of the isfinalaccess boolean flag altogether and only have this version

Reply:
That also makes a lot of sense. I'm a little worried that would introduce unexpected regressions, but it's probably worth the effort to do that now rather than later.